### PR TITLE
[LL] Make DX12 ~~great~~ compile again!

### PR DESCRIPTION
**Pull request by kvark**
This is a big WIP work, but I think we can merge this at any point in order to minimize friction with other changes.

### CODE REVIEW

File: src/backend/dx12ll/src/data.rs [hunk 1]
```diff
@@ -129,40 +129,36 @@ pub fn map_format(format: Format, is_target: bool) -> Option<DXGI_FORMAT> {
...
-    }
-    if resource_state.contains(memory::RESOLVE_DST) {
-        state = state | D3D12_RESOURCE_STATE_RESOLVE_DEST;
-    }
 
-    if resource_state.contains(memory::PRESENT) {
-        state = state | D3D12_RESOURCE_STATE_PRESENT;
+    match layout {
+        ImageLayout::Present => {
+            state = state | D3D12_RESOURCE_STATE_PRESENT;
```
Comment (msiglreith) [inline src/backend/dx12ll/src/data.rs]: `D3D12_RESOURCE_STATE_PRESENT` is _not_ a flag, is actually 0!
https://msdn.microsoft.com/de-de/library/windows/desktop/dn986744(v=vs.85).aspx

Comment (kvark) [inline src/backend/dx12ll/src/data.rs]: nicely spotted:)

**Comment by kvark** [comment]
Fixed now

**Comment by kvark** [comment]
@homu r=msiglreith

**Comment by kvark** [comment]
Oh snap, dxguid fails to compile on AppVeyor due to OOM. Might be related to the compiler bug folks has been discussing on IRC (cc @torkleyy).

**Comment by torkleyy** [comment]
It is, it's a regression in nightly / beta. It's talked through as we speak AFAIK.

**Pull request merged**